schema/logger_provider.json

```diff
@@ -65,11 +65,14 @@
             "minProperties": 1,
             "maxProperties": 1,
             "properties": {
-                "otlp": {
-                    "$ref": "common.json#/$defs/Otlp"
+                "otlp_http": {
+                    "$ref": "common.json#/$defs/OtlpHttpExporter"
+                },
+                "otlp_grpc": {
```

Comment:
There's a question of whether we should align with the names of exporters in the collector, which follows the same strategy of splitting the exporter by transport:
- [otlp](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlpexporter) for gRPC
- [otlphttp](https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/otlphttpexporter) for HTTP (proto and json)

I've chosen to _not_ align with the collector for a couple of reasons:
1. The collector follows a different case naming convention than this project. We [prefer lower snake case](https://github.com/open-telemetry/opentelemetry-configuration?tab=readme-ov-file#property-name-case), while the collector prefers something that I've seen called [flatcase or memblecase](https://stackoverflow.com/questions/17326185/what-are-the-different-kinds-of-cases). We'd have to break out case policy to align with the collector.
2. The collector prefers gRPC, which is an out of date default. By omitting a suffix on the grpc variant (i.e. `otlp` vs `otlpgrpc` and `otlphttp` instead of `otlp`), the collector is expressing a subtle preference for grpc. This was the original default for OTLP but OTEL has since switched course preferring HTTP. In the naming convention I've chosen, HTTP and gRPC are presented as equals. I don't think we should express a preference, but if we did, it should be for HTTP (i.e. `otlp` for HTTP and `otlpgrpc` for gRPC). This would be confusing for collector users, so better to explicitly break rank.